Sources/AccessibilitySnapshot/Core/Swift/Classes/HitTargetSnapshotView.swift

```diff
@@ -111,32 +123,55 @@ public enum HitTargetSnapshotUtility {
+            // less than the desired stop (small enough to be less than the density of any screen in the foreseeable
+            // future), we can avoid this rounding problem.
+            let stopEpsilon: CGFloat = 0.0001
+
             func scanLine(y: CGFloat) -> ScanLine {
                 var scanLine: ScanLine = []
-                var lastHit: (CGFloat, UIView?)? = nil
+                var lastHit: (CGFloat, UIView?) = (
+                    bounds.minX,
+                    view.hitTest(CGPoint(x: bounds.minX + touchOffset, y: y), with: nil)
+                )
+
+                func updateForHit(_ hitView: UIView?, at x: CGFloat) {
+                    if hitView == lastHit.1 {
+                        // We're still hitting the same view. Nothing to update.
+                        return
+
+                    } else {
+                        // We've moved on to a new view, so draw the scan line for the previous view.
+                        scanLine.append(((lastHit.0...x), lastHit.1))
+                        lastHit = (x, hitView)
+
+                    }
+                }
 
                 // Step through every pixel along the X axis.
-                for x in stride(from: bounds.minX, to: bounds.maxX, by: pixelWidth) {
+                for x in stride(from: bounds.minX, to: bounds.maxX, by: maxPermissibleMissedRegionWidth) {
                     let hitView = view.hitTest(CGPoint(x: x + touchOffset, y: y), with: nil)
 
-                    if let lastHit = lastHit, hitView == lastHit.1 {
+                    if hitView == lastHit.1 {
                         // We're still hitting the same view. Keep scanning.
                         continue
 
-                    } else if let previousHit = lastHit {
-                        // We've moved on to a new view, so draw the scan line for the previous view.
-                        scanLine.append(((previousHit.0...x), previousHit.1))
-                        lastHit = (x, hitView)
-
                     } else {
-                        // We've started a new view's region.
-                        lastHit = (x, hitView)
+                        // The last iteration of the loop hit test at (x - maxPermissibleMissedRegionWidth), so we want
+                        // to start one pixel in front of that.
+                        let startX = x - maxPermissibleMissedRegionWidth + pixelWidth
+
+                        for stepX in stride(from: startX, through: x, by: pixelWidth) {
+                            let stepHitView = view.hitTest(CGPoint(x: stepX + touchOffset, y: y), with: nil)
+                            updateForHit(stepHitView, at: stepX)
+                        }
                     }
                 }
 
                 // Finish the scan line if necessary.
-                if let lastHit = lastHit, let lastHitView = lastHit.1 {
-                    scanLine.append(((lastHit.0...bounds.maxX), lastHitView))
+                if lastHit.0 != bounds.maxX {
```

Comment:
Oh wait, this has a similar problem to the Y axis. This assumes the `maxPermissibleMissedRegionWidth` will get you very close to the edge of the `bounds`, but if the width isn't nicely divisible by the step size this might not be true. I think we need another stride here to finish off the pass.

Reply:
Vertical scroll indicator might be a good example of where this could fail right now